time-y1-nozzle.sh

```diff
@@ -3,79 +3,164 @@
-# requires corresponding public key in
-# https://github.com/illinois-ceesd/timing/settings/keys/new
-ssh-add timing-key.pub
-
-git clone git@github.com:illinois-ceesd/timing.git
+DRIVER_HASH=$(git rev-parse main)
+sed -e 's/\(nviz = \).*/\11000/g' \
+    -e 's/\(nrestart = \).*/\11000/g' \
+    -e 's/\(current_dt = \).*/\15e-8/g' \
+    -e 's/\(t_final = \).*/\11e-6/g' \
+    -e 's/y0_euler/nozzle-timing/g' \
+    -e 's/y0euler/nozzle-timing/g' \
+    -e 's/mode="wu"/mode="wo"/' \
+    -e 's/\(casename = \).*/\1"nozzle-timing"/g' < ./nozzle.py > ./nozzle_timing.py
+
+# -- Get an MD5Sum for the untracked nozzle_timing driver
+DRIVER_MD5SUM="None"
+if command -v md5sum &> /dev/null
+then 
+    DRIVER_MD5SUM=$(md5sum ./nozzle_timing.py | cut -d " " -f 1)
+else
+    echo "Warning: No md5sum command found. Skipping  md5sum for untracked driver."
+fi
+
+# -- Run the case (platform-dependent)
+echo RUNNING
+case $TIMING_HOST in
+
+    # --- Run the timing test in a batch job on Lassen@LLC
+    lassen*)
+        printf "Host: Lassen\n"
+        rm -f nozzle_timing_job.sh
+        rm -f timing-run-done
+        # ---- Generate a batch script for running the timing job
+        cat <<EOF > nozzle_timing_job.sh
+#!/bin/bash
+#BSUB -nnodes 1
+#BSUB -G uiuc
+#BSUB -W 30
+#BSUB -q pdebug
+
+printf "Running with EMIRGE_HOME=${EMIRGE_HOME}\n"
+
+source "${EMIRGE_HOME}/config/activate_env.sh"
+export PYOPENCL_CTX="port:tesla"
+export XDG_CACHE_HOME="/tmp/$USER/xdg-scratch"
+
+rm -f timing-run-done
+jsrun -g 1 -a 1 -n 1 python -O -u -m mpi4py ./nozzle_timing.py
+touch timing-run-done
 
-cat timing/y1-nozzle-timings.txt >> CEESD-Y1_nozzle/startup/case-time.txt
-(cd timing && git commit -m "Automatic commit $(date)" && git push)
+EOF
+        chmod +x nozzle_timing_job.sh
+        # ---- Submit the batch script and wait for the job to finish
+        bsub nozzle_timing_job.sh
+        # ---- Wait 2 minutes right off the bat (the job is at least 90 sec)
+        sleep 120
+        iwait=0
+        while [ ! -f ./timing-run-done ]; do 
+            iwait=$((iwait+1))
+            if [ "$iwait" -gt 360 ]; then # give up after 1 hour
+                printf "Timed out waiting on batch job.\n"
+                exit 1 # skip the rest of the script
+            fi
+            sleep 10
+        done
+        ;;
+
+    # --- Run the timing test on an unknown/generic machine 
+    *)
+        printf "Host: Unknown\n"
+        PYOPENCL_TEST=port:pthread python -m mpi4py ./nozzle_timing.py
+        ;;
+esac
+
+# -- Process the results of the timing run
+if [[ -f "nozzle-timing.sqlite-rank0" ]]; then
+
+    rm -f nozzle_timings.yaml
+
+    # --- Pull the timings out of the SQLITE files generated by logging
+    STARTUP_TIME=`sqlite3 nozzle-timing.sqlite-rank0 'select SUM(value) from t_step WHERE step BETWEEN 0 and 0;'`
+    FIRST_10_STEPS=`sqlite3 nozzle-timing.sqlite-rank0 'select SUM(value) from t_step WHERE step BETWEEN 0 and 10;'`
```

Comment:
```suggestion
    FIRST_10_STEPS=`sqlite3 nozzle-timing.sqlite-rank0 'select SUM(value) from t_step WHERE step BETWEEN 1 and 10;'`
```

The `BETWEEN` values are inclusive, so that second datapoint would be 11 time steps not 10.

Reply:
There should not be a timestep 0.  Timestep 0 is an actual thing, and we don't have one (yet).  If we do, then this should read BETWEEN 0 and 9, right?

Reply:
I think so, yes.